mesa/model.py

```diff
@@ -77,25 +88,66 @@ def read_csv_city(self, filename: str):
         """
         return pd.read_csv(os.path.join(self.data_folder, filename + ".csv"))
 
+    def change_global_gender_ratio(self, target_ratio):
+        ratio_m = []
+        males_pop = 0
+        females_pop = 0
+        for i in self.areas:
+            details_list = self.population_details[i][i]
+            for detail in details_list:
+                if detail[2]:  # male
+                    males_pop += detail[5]
+                else:
+                    females_pop += detail[5]
+            ratio_m.append(males_pop / (males_pop + females_pop))
+        new_list = []
+        for i in self.areas:
+            details_list = self.population_details[i][i]
+            for detail in details_list:
+                if detail[2]:
+                    detail[5] = detail[5] * target_ratio / ratio_m[i - 1]
+                else:
+                    detail[5] = detail[5] * (1 - target_ratio) / ratio_m[i - 1]
+                new_list.append(detail)
+            self.population_details[i] = {i: new_list}
```

Comment:
this data structure appears to be composed as follows:
{area#1: {area#1:{array([2, False, False, True, 78, 243.14322149749097], dtype=object)...}},
area#2: {area#2:{array([2, False, False, True, 78, 243.14322149749097], dtype=object)...},
area#2: {area#2:{array([2, False, False, True, 78, 243.14322149749097], dtype=object)...},
area#2: {area#2:{array([2, False, False, True, 78, 243.14322149749097], dtype=object)...}}

as you see there are 2 levels of repeated keys: **area#1: {area#1:{array** this looks like an error. In addition, the arrays at the last level are composed of bool, int and float. This is not good because almost no numpy function works with different types of objects. better to use a list.

Reply:
risolto